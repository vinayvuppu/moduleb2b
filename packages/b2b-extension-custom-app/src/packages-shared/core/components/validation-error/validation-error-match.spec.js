import React from 'react';
import { shallow } from 'enzyme';
import Match from './validation-error-match';

const ChildComponent = () => <div />;
ChildComponent.displayName = 'ChildComponent';

const createTestProps = custom => ({
  rule: 'rule-foo',
  isTouched: true,
  hasError: false,

  ...custom,
});

describe('<Match>', () => {
  describe('rendering', () => {
    let props;
    let wrapper;

    describe('without `error` for `rule` ', () => {
      beforeEach(() => {
        props = createTestProps();
        wrapper = shallow(
          <Match {...props}>
            <ChildComponent />
          </Match>
        );
      });

      it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
      });

      it('should not render `children`', () => {
        expect(wrapper).not.toRender(ChildComponent);
      });
    });

    describe('with `error` for `rule` ', () => {
      beforeEach(() => {
        props = createTestProps({
          hasError: true,
        });
      });
      describe('when `isTouched`', () => {
        describe('with `children`', () => {
          beforeEach(() => {
            wrapper = shallow(
              <Match {...props}>
                <ChildComponent />
              </Match>
            );
          });

          it('should match snapshot', () => {
            expect(wrapper).toMatchSnapshot();
          });

          it('should render `children`', () => {
            expect(wrapper).toRender(ChildComponent);
          });
        });

        describe('with `component`', () => {
          beforeEach(() => {
            wrapper = shallow(<Match {...props} component={ChildComponent} />);
          });

          it('should match snapshot', () => {
            expect(wrapper).toMatchSnapshot();
          });

          it('should render `children`', () => {
            expect(wrapper).toRender(ChildComponent);
          });
        });
      });
    });

    describe('when not `isTouched', () => {
      beforeEach(() => {
        props = createTestProps({
          hasError: true,
          isTouched: false,
        });
        wrapper = shallow(
          <Match {...props}>
            <ChildComponent />
          </Match>
        );
      });

      it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
      });

      it('should not render `children`', () => {
        expect(wrapper).not.toRender(ChildComponent);
      });
    });
  });
});
