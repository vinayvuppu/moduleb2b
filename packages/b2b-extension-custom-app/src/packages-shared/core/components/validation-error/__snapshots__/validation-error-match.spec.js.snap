// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Match> rendering when not \`isTouched should match snapshot 1`] = `""`;

exports[`<Match> rendering with \`error\` for \`rule\`  when \`isTouched\` with \`children\` should match snapshot 1`] = `<ChildComponent />`;

exports[`<Match> rendering with \`error\` for \`rule\`  when \`isTouched\` with \`component\` should match snapshot 1`] = `<ChildComponent />`;

exports[`<Match> rendering without \`error\` for \`rule\`  should match snapshot 1`] = `""`;
